naptime/src/main/scala/org/coursera/naptime/ETag.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2016 Coursera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.coursera.naptime
+
+import org.coursera.common.stringkey.StringKeyFormat
+import org.coursera.common.stringkey.StringKeyFormat.Implicits.OrFormat
+
+sealed trait ETag
+
+object ETag {
+
+  implicit val stringKeyFormat: StringKeyFormat[ETag] = {
+    StringKeyFormat.unimplementedFormat[ETag]
+      .orFormat[Strong]
+      .orFormat[Weak]
+  }
+
+  /**
+   * TODO(josh): Should Naptime support this at all? Play JSON serialization doesn't allow
```

Comment:
Could you actually just switch everything to use the weak etags? I believe that is the most semantically correct thing to do.